src/vs/platform/languagePacks/common/languagePacks.ts

```diff
@@ -3,10 +3,84 @@
  *  Licensed under the MIT License. See License.txt in the project root for license information.
  *--------------------------------------------------------------------------------------------*/
 
+import { CancellationTokenSource } from 'vs/base/common/cancellation';
+import { Disposable } from 'vs/base/common/lifecycle';
+import { language } from 'vs/base/common/platform';
+import { IQuickPickItem } from 'vs/base/parts/quickinput/common/quickInput';
+import { localize } from 'vs/nls';
+import { IExtensionGalleryService, IGalleryExtension } from 'vs/platform/extensionManagement/common/extensionManagement';
 import { createDecorator } from 'vs/platform/instantiation/common/instantiation';
 
 export const ILanguagePackService = createDecorator<ILanguagePackService>('languagePackService');
+
+export interface ILanguagePackItem extends IQuickPickItem {
+	readonly extensionId: string;
+	readonly galleryExtension?: IGalleryExtension;
+}
+
 export interface ILanguagePackService {
 	readonly _serviceBrand: undefined;
-	getInstalledLanguages(): Promise<string[]>;
+	getAvailableLanguages(): Promise<Array<ILanguagePackItem>>;
+	getInstalledLanguages(): Promise<Array<ILanguagePackItem>>;
+}
+
+export abstract class LanguagePackBaseService extends Disposable implements ILanguagePackService {
```

Comment:
this is abstract in common because it will also be used in a `browser` implementation in the next PR